api/log_test.go

```diff
@@ -0,0 +1,64 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"testing"
+)
+
+type logTestCase struct {
+	name       string
+	panicInput interface{}
+}
+
+func TestLogWithCheckingPanic(t *testing.T) {
+	t.Parallel()
+	testCases := []*logTestCase{
```

Comment:
nit: it can be an anonymous struct here.
```suggestion
	testCases := []struct{
		name  string
		input interface{}
		want  string
	}
```